src/exponentials.h

```diff
@@ -0,0 +1,333 @@
+/**
+ * @file exponentials.h
+ * @brief Math approximations for computing exponentials.
+ * @date January 16, 2019
+ * @author Guillaume Giudicelli, MIT, Course 22 (g_giud@mit.edu)
+ */
+
+#ifndef EXPONENTIALS_H_
+#define EXPONENTIALS_H_
+
+/**
+ * @brief Computes most of the F1 exponential term.
+ * @details This method computes (1-exp(-x))/x using a 7th order polynomial
```

Comment:
Any excellent implemented tools for Remez's algorithm? I found one, but failed to compile it. https://github.com/samhocevar/lolremez

Reply:
@cjosey implemented his own as far as I know

Reply:
Sorry for the slow reply.

I used Remez part of the way.  The precise order was:
1. Use [Boost Math](https://github.com/boostorg/math/blob/develop/minimax) minimax code (Remez) to get initial guess.
2. Use my own nonlinear optimization to tune the parameters

If you have access to Mathematica (I no longer do), the easiest I've found to work with is the [Function Approximation](https://reference.wolfram.com/language/FunctionApproximations/tutorial/FunctionApproximations.html) package.  It's significantly better than anything else I've used.